tests/unit/id.cpp

```diff
@@ -122,6 +122,68 @@ void grb_program1( const struct input &in, struct output &out ) {
 		return;
 	}
 
+
+	/**
+	 * Test for copy assignement id
+	 *
+	 * Creating and performing copy assignment on multiple new objects and check
+	 * for collisions.
+	 */
+	for( int i = 0; i < 100; i++ ) {
+		grb::Vector< int > new_one( 10 );
+		grb::Vector< int > new_two( 10 );
+		if( grb::getID( new_one ) == grb::getID( new_two ) ) {
+			std::cerr << "\t two calls to getID on two new containers return same ID\n";
+			rc = grb::FAILED;
+			return;
+		}
+		const size_t old_one_id = grb::getID( new_one );
+		const size_t old_two_id = grb::getID( new_two );
+		new_two = new_one;
+
+		if( grb::getID( new_two ) == old_one_id ) {
+			std::cerr << "\t detected duplicated ID transferred with copy-assignment\n";
+			rc = grb::FAILED;
+			return;
+		}
+		if( (grb::getID( new_two ) != old_two_id)
+         || (grb::getID( new_one ) == grb::getID( new_two ) ) ) {
+			std::cerr << "\t detected new ID was different from original after assignment\n";
+			rc = grb::FAILED;
+			return;
+		}
+	}
+
+	/**
+	 * Test for move assignement id
+	 *
+	 * Creating and performing copy assignment on multiple new objects and check
+	 * for collisions.
+	 */
+	for( int i = 0; i < 100; i++ ) {
+		grb::Vector< int > new_one( 10 );
+		grb::Vector< int > new_two( 10 );
+		if( grb::getID( new_one ) == grb::getID( new_two ) ) {
+			std::cerr << "\t two calls to getID on two new containers return same ID\n";
+			rc = grb::FAILED;
+			return;
+		}
+		const size_t old_one_id = grb::getID( new_one );
+		const size_t old_two_id = grb::getID( new_two );
+		new_two = std::move( new_one );
+
+		if( grb::getID( new_two ) == old_one_id ) {
```

Comment:
I think `old_one_id` should be `old_two_id`. Rationale: the `std::move` is to pass all resources underlying `new_two` to `new_one`. Thus both practically and semantically, `new_two` "becomes" `new_one` (while `new_one` enters some undefined unusable state). Therefore, if `new_two` still has the same ID as before, then something went wrong -- while the expectation in fact is that `new_two` has the former ID of `new_one`. This also matches the logic for the test later on in this file, for matrices rather than vectors.